core/src/main/scala/com/raphtory/formats/CsvFormat.scala

```diff
@@ -39,37 +39,36 @@ case class CsvFormat(delimiter: String = ",") extends Format {
   ): SinkExecutor =
     new SinkExecutor {
       var currentPerspective: Perspective = _
-      private val mapper       = JsonMapper.builder().addModule(DefaultScalaModule).build()
+      private val mapper                  = JsonMapper.builder().addModule(DefaultScalaModule).build()
 
-      private def ensureQuoted(str: String): String = {
-        if ((str.startsWith("\"") && str.endsWith("\""))
-          || (str.startsWith("'") && str.endsWith("'" ))
-          || !str.contains(delimiter)) {
+      private def ensureQuoted(str: String): String =
+        if (
+                (str.startsWith("\"") && str.endsWith("\""))
+                || (str.startsWith("'") && str.endsWith("'"))
+                || !str.contains(delimiter)
+        )
           str
-        } else {
+        else
           "\"" + str + "\""
-        }
-      }
-
 
-
-      private def csvValue(obj: Any): String = {
+      private def csvValue(obj: Any): String =
         obj match {
           case v: String => ensureQuoted(v)
-          case v =>
+          case v         =>
             ensureQuoted(mapper.writeValueAsString(v))
         }
-      }
 
       override def setupPerspective(perspective: Perspective): Unit =
         currentPerspective = perspective
 
       override protected def writeRow(row: Row): Unit = {
-        val value = currentPerspective.window match {
+        val timestamp =
+          if (currentPerspective.formatAsDate) currentPerspective.timestampAsDatetime else currentPerspective.timestamp
+        val value     = currentPerspective.window match {
           case Some(w) =>
-            s"${currentPerspective.timestamp}$delimiter$w$delimiter${row.getValues().map(csvValue).mkString(delimiter)}"
+            s"$timestamp$delimiter$w$delimiter${row.getValues().map(csvValue).mkString(delimiter)}"
           case None    =>
-            s"${currentPerspective.timestamp}$delimiter${row.getValues().map(csvValue).mkString(delimiter)}"
+            s"$timestamp$delimiter${row.getValues().map(csvValue).mkString(delimiter)}"
         }
```

Comment:
once `timestampAsString` is implemented as per above, this can just call it to get the appropriate
representation